src/acktest/bootstrapping/iam.py

```diff
@@ -73,18 +121,25 @@ def bootstrap(self):
     def cleanup(self):
         """Deletes an IAM role.
         """
-        role_name = re.match(ROLE_ARN_REGEX, self.arn).group(1)
-        managed_policy = self.iam_client.list_attached_role_policies(RoleName=role_name)
-        for each in managed_policy["AttachedPolicies"]:
-            self.iam_client.detach_role_policy(RoleName=role_name, PolicyArn=each["PolicyArn"])
-
-        inline_policy = self.iam_client.list_role_policies(RoleName=role_name)
-        for each in inline_policy["PolicyNames"]:
-            self.iam_client.delete_role_policy(RoleName=role_name, PolicyName=each)
-
-        instance_profiles = self.iam_client.list_instance_profiles_for_role(RoleName=role_name)
-        for each in instance_profiles["InstanceProfiles"]:
-            self.iam_client.remove_role_from_instance_profile(
-                RoleName=role_name, InstanceProfileName=each["InstanceProfileName"]
-            )
-        self.iam_client.delete_role(RoleName=role_name)
\ No newline at end of file
+        if self.arn:
+            role_name = re.match(ROLE_ARN_REGEX, self.arn).group(1)
+            
+            managed_policy = self.iam_client.list_attached_role_policies(RoleName=role_name)
+            for each in managed_policy["AttachedPolicies"]:
+                self.iam_client.detach_role_policy(RoleName=role_name, PolicyArn=each["PolicyArn"])
+
+            inline_policy = self.iam_client.list_role_policies(RoleName=role_name)
+            for each in inline_policy["PolicyNames"]:
+                self.iam_client.delete_role_policy(RoleName=role_name, PolicyName=each)
```

Comment:
don't need to detach these before delete?

Reply:
`list_role_policies` lists the inline policies for the role, so they only need to be deleted.